src/MiniProfiler.Shared/ClientTimings.cs

```diff
@@ -14,12 +14,26 @@ public class ClientTimings
     {
         private const string TimingPrefix = "clientPerformance[timing][";
         private const string ProbesPrefix = "clientProbes[";
+        private readonly object _lockObject = new object();
+        private List<ClientTiming> _timings;
 
         /// <summary>
         /// Gets or sets the list of client side timings
         /// </summary>
         [DataMember(Order = 2)]
-        public List<ClientTiming> Timings { get; set; }
+        public List<ClientTiming> Timings
+        {
+            get
+            {
+                lock(_lockObject)
+                    return _timings == null ? null : new List<ClientTiming>(_timings);
```

Comment:
This creates a duplicate list *on every access*, that's a lot of allocations for the safety.